homeassistant/components/nordpool/sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Support for Nordpool electrical prices sensors."""
+import logging
+
+from datetime import (datetime, timedelta)
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (ATTR_ATTRIBUTION,
+                                 CONF_NAME,
+                                 CONF_CURRENCY,
+                                 CONF_REGION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['nordpool==0.2']
+
+SCAN_INTERVAL = timedelta(minutes=60)
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:cash-usd'
+ATTRIB = "For details, see https://nordpoolgroup.com/"
+
+CURRENCY_FRACTION = {
+    'DKK': 'Øre',
+    'EUR': 'Cent',
+    'NOK': 'Øre',
+    'SEK': 'Øre'
+}
+
+VALID_REGION = ['DK1',
+                'DK2',
+                'EE',
+                'FI',
+                'LT',
+                'LV',
+                'Oslo',
+                'Kr.sand',
+                'Bergen',
+                'Molde',
+                'Tr.heim',
+                'Tromsø',
+                'SE1',
+                'SE2',
+                'SE3',
+                'SE4',
+                'SYS']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_CURRENCY): vol.All(cv.string, vol.In(CURRENCY_FRACTION)),
+    vol.Required(CONF_REGION): vol.All(cv.string, vol.In(VALID_REGION)),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Nordpool sensor platform."""
+    sens_name = config[CONF_NAME]
+    sens_currency = config[CONF_CURRENCY]
+    sens_region = config[CONF_REGION]
+
+    sens_nordpool = NordpoolAPI(sens_name, sens_currency, sens_region)
+    add_entities([sens_nordpool], True)
+
+
+class NordpoolAPI(Entity):
+    """Initiate communications to Nordpool API."""
+
+    def __init__(self, sens_name, sens_currency, sens_region):
+        """Init new sensor."""
+        self._name = sens_name
+        self._unit_of_measurement = CURRENCY_FRACTION[sens_currency]
+        self._region = sens_region
+        self._state = None
+        self._currency = sens_currency
+        self._day = None
+        self._data = None
+        self._device_state_attributes = {}
+
+    @Throttle(SCAN_INTERVAL)
```

Comment:
The throttle will only set the lower limit of the configurable scan interval. Ie it will throttle per entity, not per home assistant. Is that ok?

Eg two nordpol platforms set up would be able to each scan within the same hour.

Reply:
So if I remove the Throttle line, HA will automatically update as per SCAN_INTERVAL?

I was under the impression that to be able to control the update interval I needed to insert a throttle.

Reply:
Yes, home assistant will use the `SCAN_INTERVAL` constant if defined to set the default scan interval for platforms.

The throttle is mostly useful if there's a single data fetching instance that all entities poll and we want throttle the data fetching to a certain interval.

Reply:
Okay - will remove throttle then.